test/e2e/tests/confirmations/contract-interaction-redesign.spec.js

```diff
@@ -0,0 +1,144 @@
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.clickElement(`#deployButton`);
+
+        await driver.delay(1000);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Contract deployment',
+        });
+
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        await driver.clickElement({ text: 'Activity', tag: 'button' });
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .activity-list-item:nth-of-type(1)',
+        );
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.clickElement(`#depositButton`);
+
+        await driver.delay(1000);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        await driver.waitForSelector({
+          css: 'h2',
+          text: 'Transaction request',
+        });
+      },
+    );
+  });
+
+  it(`Opens a contract interaction type 2 transaction (EIP1559)`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptionsForType2Transactions,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+        await driver.clickElement(`#deployButton`);
+
+        await driver.delay(1000);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Contract deployment',
+        });
+
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+        await driver.clickElement({ text: 'Activity', tag: 'button' });
+        await driver.waitForSelector(
+          '.transaction-list__completed-transactions .activity-list-item:nth-of-type(1)',
+        );
```

Comment:
same as before, we can skip all the deploy contract steps and use fixtures+ganache seeder